lib/ansible/modules/cloud/google/gcp_compute_image.py

```diff
@@ -0,0 +1,753 @@
+        u'imageEncryptionKey': ImageImageEncryKey(response.get(u'imageEncryptionKey', {}), module).from_response(),
+        u'licenses': response.get(u'licenses'),
+        u'name': response.get(u'name'),
+        u'rawDisk': ImageRawDisk(response.get(u'rawDisk', {}), module).from_response(),
+        u'sourceDisk': response.get(u'sourceDisk'),
+        u'sourceDiskEncryptionKey': ImagSourDiskEncrKey(response.get(u'sourceDiskEncryptionKey', {}), module).from_response(),
+        u'sourceDiskId': response.get(u'sourceDiskId'),
+        u'sourceType': response.get(u'sourceType')
+    }
+
+
+def async_op_url(module, extra_data=None):
+    if extra_data is None:
+        extra_data = {}
+    url = "https://www.googleapis.com/compute/v1/projects/{project}/global/operations/{op_id}"
+    combined = extra_data.copy()
+    combined.update(module.params)
+    return url.format(**combined)
+
+
+def wait_for_operation(module, response):
+    op_result = return_if_object(module, response, 'compute#operation')
+    if op_result is None:
+        return None
+    status = navigate_hash(op_result, ['status'])
+    wait_done = wait_for_completion(status, op_result, module)
+    return fetch_resource(module, navigate_hash(wait_done, ['targetLink']), 'compute#image')
+
+
+def wait_for_completion(status, op_result, module):
+    op_id = navigate_hash(op_result, ['name'])
+    op_uri = async_op_url(module, {'op_id': op_id})
+    while status != 'DONE':
+        raise_if_errors(op_result, ['error', 'errors'], 'message')
+        time.sleep(1.0)
+        if status not in ['PENDING', 'RUNNING', 'DONE']:
+            module.fail_json(msg="Invalid result %s" % status)
+        op_result = fetch_resource(module, op_uri, 'compute#operation')
+        status = navigate_hash(op_result, ['status'])
+    return op_result
+
+
+def raise_if_errors(response, err_path, module):
+    errors = navigate_hash(response, err_path)
+    if errors is not None:
+        module.fail_json(msg=errors)
+
+
+class ImageDeprecated(object):
+    def __init__(self, request, module):
+        self.module = module
+        if request:
+            self.request = request
+        else:
+            self.request = {}
+
+    def to_request(self):
+        return remove_nones_from_dict({
+            u'deleted': self.request.get('deleted'),
+            u'deprecated': self.request.get('deprecated'),
+            u'obsolete': self.request.get('obsolete'),
+            u'replacement': self.request.get('replacement'),
+            u'state': self.request.get('state')
+        })
+
+    def from_response(self):
+        return remove_nones_from_dict({
+            u'deleted': self.request.get(u'deleted'),
+            u'deprecated': self.request.get(u'deprecated'),
+            u'obsolete': self.request.get(u'obsolete'),
+            u'replacement': self.request.get(u'replacement'),
+            u'state': self.request.get(u'state')
+        })
+
+
+class ImageGuestOsFeatuArray(object):
```

Comment:
Can you allow longer class names in the generator, so this can spell out as `ImageGuestOsFeatureArray`?